txscript/sigcache.go

```diff
@@ -6,22 +6,36 @@
 package txscript
 
 import (
+	"crypto/rand"
+	"encoding/binary"
 	"sync"
 
+	"github.com/dchest/siphash"
 	"github.com/decred/dcrd/chaincfg/chainhash"
 	"github.com/decred/dcrd/dcrec/secp256k1/v3"
 	"github.com/decred/dcrd/dcrec/secp256k1/v3/ecdsa"
+	"github.com/decred/dcrd/wire"
 )
 
+// ProactiveEvictionDepth is the depth of the block at which the signatures for
+// the transactions within the block are nearly guaranteed to no longer be
+// useful.
+const ProactiveEvictionDepth = 2
```

Comment:
Shouldn't this be just after the maximum possible reorg depth?

Reply:
That would make the cache much less effective since it primarily intended to prevent the need to recheck signatures that were just checked when the transactions entered the mempool.